app.js

```diff
@@ -4,15 +4,108 @@ const express = require('express');
 const hbs = require('hbs');
 
 // require spotify-web-api-node package here:
+const SpotifyWebApi = require('spotify-web-api-node');
 
 const app = express();
 
+const port = 3010;
 app.set('view engine', 'hbs');
 app.set('views', __dirname + '/views');
 app.use(express.static(__dirname + '/public'));
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => {
+    console.log('The access token expires in ' + data.body['expires_in']);
+    console.log('The access token is ' + data.body['access_token']);
+    return spotifyApi.setAccessToken(data.body['access_token']);
+  })
+  .catch((error) => {
+    console.log('Something went wrong when retrieving an access token', error);
+    if (error.statusCode == 401) {
+      console.log('**** ACCESS TOKEN EXPIRED ***** AND NOW? ****');
+    }
+    return;
+  });
 
 // Our routes go here:
+app.get('/', (req, res) => {
+  res.render('index');
+});
+app.get('/artist-search', (req, res) => {
+  spotifyApi
+    .searchArtists(req.query.q, { limit: 20, offset: 0 })
+    .then((data) => {
+      console.log('artist-search searchArtists');
+
+      if (req.query.help && req.query.help == 1) {
+        res.send(data.body);
```

Comment:
I am curious what you are trying to do here :)

Reply:
by doing this wherever I am, I can changed the URL and add ?help=1 to it and it will show me the JSON returned by Spotify instead of the nice page.  ;-)